Add git-lfs to setup process

AramZS opened the pull request:
This PR will add the installation of [git-lfs](https://git-lfs.github.com/) to the container and initiate it during the initial setup of git during the RUN phase, allowing the container to handle the use of large files pushed to the Github Pages site and branch.  

This addresses #748 by applying the right machine-level systems but does not completely solve the problem, as the user will still need to find a way to pass their `.gitattributes` file to the base of their new site the same way they might need to with a `CNAME` file.

AramZS committed:
Add git-lfs to setup.

peaceiris commented:
This action is a TypeScript action.

AramZS commented:
> This action is a TypeScript action.

What do you mean @peaceiris? LFS is configured at the Git level and Typescript shouldn't be required. Is there a TypeScript definition file that I need to add information to somewhere?

peaceiris commented:
[About custom actions - GitHub Docs](https://docs.github.com/en/actions/creating-actions/about-custom-actions)

AramZS committed:
Remove mistaken Docker command, use git-lfs in setup

This change sets up Git LFS using the same approach as @actions/checkout
and will hopefully allow the use of Git LFS to match the potential use
in the checkout action and allow users to push up published GitHub
pages using `git lfs` to support files larger than 100MB.

AramZS committed:
Add test coverage for LFS input

AramZS commented:
@peaceiris Ah my apologies, I was unclear on that. I haven't done much work on Actions and so am somewhat unfamiliar with the whole process. Forgive my misunderstanding here. Thank you for clarifying. 

I've pushed up additional changes and accompanying documentation, along with a flag that a user can use to turn it on instead of turning it on by default. I've mirrored how it is set up in [@actions/checkout](https://github.com/actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/git-command-manager.ts#L255) with the idea that the developer would be using that flag and this one.

I am not entirely sure if I've set it up correctly, everything looks like it should work in the same way as with the checkout action, I think. 

But I'm not sure how to test it locally. I'd be glad to do that work, but I'm unsure if your setup is automated or something manual for that? I'd be glad to do the work, but I think I might need advice on how to best set up such a test in the context of your project.

peaceiris commented:
Thanks!

As I mentioned in [this comment](https://github.com/peaceiris/actions-gh-pages/issues/748#issuecomment-1147528927), GitHub Pages does not support objects under Git LFS. So there is no way to **upload** files under LFS. Therefore, the `lfs` input is not needed for this action.

Adding the section about **downloading** Git LFS objects on GitHub Actions runners can be useful for users who want to commit and push the large size of files less than 100MB as Git normal objects. So I will merge this pull-request as adding the section including a recipe of the `lfs` input of actions/checkout.

AramZS commented:
@peaceiris I agree that the large objects won't be available in the pages site, but if we checkout with LFS, we should be able to check in to the branch with those files. They won't be published to the Github Pages site, but I know the files can still exist in branch used for GitHub Pages and without the addition of LFS at the time of commit in this action what will happen is that LFS will be used for checkout and when this action attempts to make a git commit for publication it will crash. So I wouldn't recommend adding the LFS documentation without support for the page-branch commit in this action, otherwise users will experiance breakage.

AramZS commented:
@peaceiris I wanted to check in on this. I think that the configuration here should work and resolve the issue, allowing git-lfs files in the built branch to be committed without crashing the process?

sojs-coder commented:
@AramZS , I am in a similar situation... reading through the code all appears to be in working order. Can we merge this? @peaceiris
